### ISSUE TITLE
Failure when trying to run job with custom suites 

### ISSUE THREAD

**Issue by clebergnu** [avocado-framework/avocado#4141]
The following is a slightly modified version of `examples/jobs/passjob_custom.py`, with the goal of eliminating the nrunner as the cause of this issue.

```python
import sys

from avocado.core.job import Job
from avocado.core.suite import TestSuite

job_config = {}

config1 = {'run.references': ['examples/tests/passtest.py:PassTest.test']}
config2 = {'run.references': ['examples/tests/passtest.py:PassTest.test']}

# Custom method (no discovery, no guess, no magic)
# Since there is no magic, we need to pass a suite name, otherwise a uuid4 will
# be used for suite.name. Also run.references will be ignored (Avocado will not
# creating tests suites for you).

suite1 = TestSuite(config=config1, tests=[], name='suite1')
suite2 = TestSuite(config=config2, tests=[], name='suite2')

with Job(job_config, [suite1, suite2]) as j:
    sys.exit(j.run())
```

This results in:

```
 $ python3 examples/jobs/passjob_custom.py 
JOB ID     : cd23d5f38093ea0deacea31a7b515ae3c269c317
JOB LOG    : /home/cleber/avocado/job-results/job-2020-08-25T22.19-cd23d5f/job.log
Unable to resolve any reference or 'run.references' is empty.
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```

Expected result is a job with 2 test suites (and thus 2 tests) executed.

**Comment by beraldoleal** [comment]
@clebergnu afaik, this is expected as we discussed, to keep consistency.

You are creating TestSuite() and Job(), from the "custom mode" (class constructors) so there will be no magic to find tests based on `run.references` on Avocado side. This expect that you create your tests by hand.

If we need some 'automatic' logic, then we should use `.from_config()` methods.

Makes sense?

**Comment by clebergnu** [comment]
> @clebergnu afaik, this is expected as we discussed, to keep consistency. And it is saying on the comment (maybe it is not a good string):
> 
> > Since there is no magic, we need to pass a suite name, otherwise a uuid4 will be used for suite.name. Also run.references will be ignored (Avocado will not creating tests suites for you).
> 
> You are creating TestSuite() and Job(), from the "custom mode" (class constructors) so there will be no magic to find tests based on `run.references` on Avocado side. This expects that you create your tests by hand.
> 
> If we need some 'automatic' logic, then we should use `.from_config()` methods.
> 
> Makes sense?

Yes, it does.  Then the problem is that the example job is misleading, as it uses `run.references` without `.from_config()`.  I will include a fix on v2 of #4142
